### PR TITLE
fix: workflow permissions

### LINKED ISSUE #17: Knip CI - II
1. Please move knip.ts into `.github/workflows` 
2. Fix the auth permissions issue ideally by adjusting the CI settings or at most by setting up a basic github app with minimal permissions to write a comment to pull requests

Related https://github.com/ubiquity/audit.ubq.fi/pull/2#issuecomment-1974745936

### PR DESCRIPTION
**Pull request by gentlementlegen**
Resolves #17 

Depends on https://github.com/ubiquity/ts-template/pull/24 (needs to be merged and rebased)

### CODE REVIEW

**Comment by FernandVEYRIER** [comment]
More context about what is going on:
On `pull-request` events, GitHub actually sets the `GITHUB_TOKEN` to read-only, to avoid external workflows to run and either break the repo, or steal private tokens etc, as a security measure. Problem is, some of our workflows require write access to work properly. I tried changing the event to `pull_request_target`, difference being that the Action runs inside the `base` repository context and not the source one. In such case GitHub does populate the token properly. I don't think I can test it without merging it first and see if the issue is solved.

More reading here:
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target
- https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

**Comment by gentlementlegen** [comment]
I don't think I can test it before it's merged, since it has to be updated on the base repository to be triggered properly.

File: tests/main.test.ts
Comment (0x4007) [inline tests/main.test.ts:1]: What's this test for

**Comment by 0x4007** [comment]
You can rebase and test.

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**
